generate/convert.go

```diff
@@ -439,6 +439,11 @@ func (g *generator) convertDefinition(
 				return nil, err
 			}
 
+			if options.GetPointer() || (!field.Type.NonNull && g.Config.Optional == "pointer") {
+				oe := true
+				fieldOptions.Omitempty = &oe
```

Comment:
There's no need to mutate `fieldOptions` -- just compute the correct value and then use it on line 453.

Reply:
A belated realization: actually the logic here is also wrong: users may need to override the global option locally, if they do need to pass an explicit null. (Sadly in Go, as bad as it is to represent "T or null", it's even worse to represent "T or null or unset", which to be fair is also true of almost any language that's not JavaScript.) That is, if you set this option, then you can do `# @genqlient(omitempty: false)` if you do want explicit null for some query.

So that means you'll need to actually look at the value of `fieldOptions.Omitempty` and then only use this logic if it's null, I think.

Reply:
@benjaminjkraft there doesn't seem to be move movement here on your feedback but I realy really need this feature so I've created https://github.com/Khan/genqlient/pull/287 with your suggestion, in order to get things forward. Feel free to take a look over there :)